### PR TITLE
feat: [ZN-388] New starter documentation

### PR DESCRIPTION
**Pull request by SkrzypMajster**
Hi guys!

I put some work into refactoring our `react-starter-boilerplate` project documentation. I expanded the `README.md` file and added the `docs` directory, which contains documents describing specific aspects of the project based on our starter. These documents include for example list of libraries used, directory structure, and many other implemented mechanisms.

I also added the `CONTRIBUTING.md` file which describes the process of contribution to the `react-starter-boilerplate` project.

### CODE REVIEW

File: PROJECT_README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- `build` - Compiles and bundles the app for deployment*
+- `build:ci` - Build command optimized for CI/CD pipelines
+- `typecheck` - Validate the code using TypeScript compiler
+- `preview` - Boot up a local static web server that serves application build. It's an easy way to check if the production build looks OK on your local machine
+- `test` - Run unit tests with vitest
+- `coverage` - Run unit tests with code coverage calculation
+- `lint` - Validate the code using ESLint and Prettier
+- `lint:fix` - Validate and fix the code using ESLint and Prettier
+- `plop` - Run CLI with commands for code generation
+- `translations` - Run Babelsheet tool for fetch latest translations
```
Comment (qh4r) [inline PROJECT_README.md:38]: mention babelsheet https://github.com/TheSoftwareHouse/babelsheet2

File: README.md [hunk 2]
```diff
@@ -1,135 +1,120 @@
...
-Launches the test runner in the coverage report generation mode.<br /> See
-[this](https://vitest.dev/guide/coverage.html) section for more information.
+Now, your project is bootstrapped successfully! 🎉
 
-### `npm run build`
+You can install dependencies and start developing your React application 🚀
 
-Builds the app for production to the `build` folder.<br /> It correctly bundles React in production mode and optimizes
-the build for the best performance.
+## Scripts
```
Comment (qh4r) [inline README.md:72]: There is quite a lot of dupliticy with other MD files

**Review by qh4r — APPROVED** [review]

File: PROJECT_README.md (hunk 1 above)
Comment (SkrzypMajster) [reply PROJECT_README.md]: mentioned ✅

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+After that, TSH team will take a look at your issue and try to figure it out.
+
+## Making Pull Requests
+
+PR's to this repository are also very welcome.
+
+If you want to resolve one of the issues reported or add new functionality to the starter project, you need first to fork this repository.
+
+Please note that we are using [conventional commits](https://www.conventionalcommits.org/) methodology of naming commits in `react-starter-boilerplate` repository.
```
[the remark this answers is not in the document]
Comment (SkrzypMajster) [reply CONTRIBUTING.md]: I updated `CONTRIBUTING.md` file a little bit to mention also branch naming pattern 🙂

**Review by qh4r — APPROVED** [review]

**Pull request merged**